satellite/gracefulexit/endpoint.go

```diff
@@ -649,78 +535,22 @@ func (endpoint *Endpoint) handleSucceeded(ctx context.Context, stream processStr
-	}
-	if transfer.satelliteMessage.GetTransferPiece() == nil {
-		return Error.New("Satellite message transfer piece cannot be nil")
-	}
-	if transfer.satelliteMessage.GetTransferPiece().GetAddressedOrderLimit() == nil {
-		return Error.New("Addressed order limit on transfer piece cannot be nil")
-	}
-	if transfer.satelliteMessage.GetTransferPiece().GetAddressedOrderLimit().GetLimit() == nil {
-		return Error.New("Addressed order limit on transfer piece cannot be nil")
-	}
-	if transfer.path == nil {
-		return Error.New("Transfer path cannot be nil")
-	}
-
-	originalOrderLimit := message.Succeeded.GetOriginalOrderLimit()
-	if originalOrderLimit == nil {
-		return ErrInvalidArgument.New("Original order limit cannot be nil")
-	}
-	originalPieceHash := message.Succeeded.GetOriginalPieceHash()
-	if originalPieceHash == nil {
-		return ErrInvalidArgument.New("Original piece hash cannot be nil")
-	}
-	replacementPieceHash := message.Succeeded.GetReplacementPieceHash()
-	if replacementPieceHash == nil {
-		return ErrInvalidArgument.New("Replacement piece hash cannot be nil")
-	}
-
-	// verify that the original piece hash and replacement piece hash match
-	if !bytes.Equal(originalPieceHash.Hash, replacementPieceHash.Hash) {
-		return ErrInvalidArgument.New("Piece hashes for transferred piece don't match")
-	}
-
-	// verify that the satellite signed the original order limit
-	err = endpoint.orders.VerifyOrderLimitSignature(ctx, originalOrderLimit)
+	err = endpoint.validatePendingTransfer(ctx, transfer)
 	if err != nil {
-		return ErrInvalidArgument.Wrap(err)
-	}
-
-	// verify that the public key on the order limit signed the original piece hash
-	err = signing.VerifyUplinkPieceHashSignature(ctx, originalOrderLimit.UplinkPublicKey, originalPieceHash)
-	if err != nil {
-		return ErrInvalidArgument.Wrap(err)
-	}
-
-	if originalOrderLimit.PieceId != message.Succeeded.OriginalPieceId {
-		return ErrInvalidArgument.New("Invalid original piece ID")
+		return Error.Wrap(err)
 	}
 
 	receivingNodeID := transfer.satelliteMessage.GetTransferPiece().GetAddressedOrderLimit().GetLimit().StorageNodeId
-	if transfer.originalPointer == nil || transfer.originalPointer.GetRemote() == nil {
-		return Error.New("could not get remote pointer from transfer item")
-	}
-	calculatedNewPieceID := transfer.originalPointer.GetRemote().RootPieceId.Derive(receivingNodeID, transfer.pieceNum)
-	if calculatedNewPieceID != replacementPieceHash.PieceId {
-		return ErrInvalidArgument.New("Invalid replacement piece ID")
-	}
-
 	// get peerID and signee for new storage node
 	peerID, err := endpoint.peerIdentities.Get(ctx, receivingNodeID)
 	if err != nil {
 		return Error.Wrap(err)
 	}
-	signee := signing.SigneeFromPeerIdentity(peerID)
-
-	// verify that the new node signed the replacement piece hash
-	err = signing.VerifyPieceHashSignature(ctx, signee, replacementPieceHash)
+	// verify transferred piece
+	err = endpoint.verifyPieceTransferred(ctx, message, transfer, peerID)
```

Comment:
nit - `peerID` should be able to be acquired from inside `verifyPieceTransferred` now that it is a function on the endpoint. Lines 543-548 should be able to be moved inside